versions.tf

```diff
@@ -1,3 +1,5 @@
 terraform {
-  required_version = ">= 0.12.9"
+  required_version = ">= 0.14"
```

Comment:
Since the new `service_type` argument of the `aws_vpc_endpoint_service` resource requires the 3.x AWS Provider it would be good to add (assuming tf 0.14 isn't adopted):

```hcl
terraform {
  required_version = ">= 0.12.26"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.0"
    }
  }
}
```
`0.12.26` added [limited support for the "required_providers" syntax](https://www.terraform.io/docs/language/providers/requirements.html#v0-12-compatible-provider-requirements).

This will make it so that anyone attempting to utilize the module on a stack locked to a 2.x provider will get an error message about incompatible provider versions.

Reply:
Reading that link:

> If the provider belongs to the hashicorp namespace, as with the hashicorp/aws provider shown above, omit the source argument and allow Terraform v0.13 to select the hashicorp namespace by default.

So I think the block should be:

```
terraform {
  required_version = ">= 0.12.26"

  required_providers {
    aws = {
      version = ">= 3.0"
    }
  }
}
```